src/FrameDecorator/AbstractFrameDecorator.php

```diff
@@ -171,13 +179,13 @@ function deep_copy()
         return $deco;
     }
 
-    /**
-     * Delegate calls to decorated frame object
-     */
     function reset()
     {
         $this->_frame->reset();
+        $this->reset_generated_content();
+        $this->revert_counter_increment();
```

Comment:
Reverting incremented counters on frame reset handles cases where a child that already had a reflow (and thus counter increment) is moved to the next page.